examples/flutter_gallery/lib/demo/video_demo.dart

```diff
@@ -0,0 +1,285 @@
+    }
+
+    return new Stack(
+      alignment: Alignment.bottomCenter,
+      fit: StackFit.passthrough,
+      children: children,
+    );
+  }
+}
+
+class FadeAnimation extends StatefulWidget {
+  final Widget child;
+  final Duration duration;
+
+  const FadeAnimation(
+      {this.child, this.duration: const Duration(milliseconds: 500)});
+
+  @override
+  _FadeAnimationState createState() => new _FadeAnimationState();
+}
+
+class _FadeAnimationState extends State<FadeAnimation>
+    with SingleTickerProviderStateMixin {
+  AnimationController animationController;
+
+  @override
+  void initState() {
+    super.initState();
+    animationController =
+        new AnimationController(duration: widget.duration, vsync: this);
+    animationController.addListener(() {
+      if (mounted) {
+        setState(() {});
+      }
+    });
+    animationController.forward(from: 0.0);
+  }
+
+  @override
+  void deactivate() {
+    animationController.stop();
+    super.deactivate();
+  }
+
+  @override
+  void didUpdateWidget(FadeAnimation oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.child != widget.child) {
+      animationController.forward(from: 0.0);
+    }
+  }
+
+  @override
+  void dispose() {
+    animationController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return animationController.isAnimating
+        ? new Opacity(
+            opacity: 1.0 - animationController.value,
+            child: widget.child,
+          )
+        : new Container();
+  }
+}
+
+class VideoDemo extends StatefulWidget {
+  const VideoDemo({Key key}) : super(key: key);
+
+  static const String routeName = '/video';
+
+  @override
+  _VideoDemoState createState() => new _VideoDemoState();
+}
+
+class _VideoDemoState extends State<VideoDemo> {
+  final VideoPlayerController butterflyController =
+      new VideoPlayerController(butterflyUri);
+  final VideoPlayerController beeController = new VideoPlayerController(beeUri);
+
+  @override
+  void initState() {
+    super.initState();
+
+    Future<Null> initController(VideoPlayerController controller) async {
+      controller.setLooping(true);
+      controller.setVolume(0.0);
+      controller.play();
+      await controller.initialize();
+      setState(() {});
+    }
+
+    initController(butterflyController);
+    initController(beeController);
+  }
+
+  @override
+  void dispose() {
+    butterflyController.dispose();
+    beeController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: const Text('Videos'),
+      ),
+      body: new ListView(
+        children: <Widget>[
+          new VideoCard(
+            controller: butterflyController,
+            title: 'Float',
+            subtitle: '… like a butterfly',
+          ),
+          new VideoCard(
+            controller: beeController,
+            title: 'Sting',
```

Comment:
This phrase can be interpreted as antagonistic to an American audience. Is there another option we could use?

Reply:
How about:

```
new VideoCard(
   title: 'Butterfly',
   subtitle: 'flutters by',
),
new VideoCard(
   title: 'Bee',
   subtitle: 'gently buzzing',
),
```

?

Reply:
sure :)

bonus points if you can write a haiku!

Reply:
Done